lib/path.js

```diff
@@ -673,7 +674,7 @@ const win32 = {
   },
 
 
-  _makeLong: function _makeLong(path) {
+  toLongUNCPath: function toLongUNCPath(path) {
```

Comment:
This name is incorrect, the result is not UNC but a specially prefixed "extended lengh" path name.
IMHO best option is `resolveExtended` (although to keep with `dirname` and `extname` is could be all lowercase `resolveextended`)
Ref: https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx
![image](https://user-images.githubusercontent.com/96947/29515550-d8648e9e-863a-11e7-9835-d6da1f92d199.png)